### PR TITLE
Fix import sample

### PR DESCRIPTION
**Pull request by chiahaoliu**
this PR fixes a subtle problem I didn't find in the ``py.test``. We need to grab ``bt`` which is currently alive but not reload an equivalent``bt`` inside function, so that ``Sample`` objects are linked properly, i.e. ``bt.list()`` in live ``ipython`` session shows updated result after importing.

@sbillinge  @CJ-Wright  this PR is ready for review

### CODE REVIEW

File: xpdacq/utils.py [hunk 1]
```diff
@@ -6,12 +6,38 @@
...
 from .beamtime import Beamtime, Sample, ScanPlan
 
+
+def _check_obj(required_obj_list):
+    """ function to check if object(s) exist
+
+    Parameter
+    ---------
+    required_obj_list : list
+        a list of strings refering to object names
```
Comment (sbillinge) [inline xpdacq/utils.py]: typo

File: xpdacq/utils.py [hunk 2]
```diff
@@ -6,12 +6,38 @@
...
+
+    """
+    ips = get_ipython()
+    for obj_str in required_obj_list:
+        if not ips.ns_table['user_global'].get(obj_str, None):
+            raise NameError("Required object {} doesn't exist in"
```
Comment (sbillinge) [inline xpdacq/utils.py]: can we have an error message that tells the user what to do?

File: xpdacq/utils.py [hunk 3]
```diff
@@ -487,21 +516,49 @@ def import_sample_info(saf_num=None, bt=None):
...
+    bt : xpdacq.beamtime.Beamtime
+        beamtime object that is going to be linked with these samples
+    """
+
+    # at core function level, bt should strictly be Beamtime,
+    # raise if it is not this case
+    if not isinstance(bt, Beamtime):
+        raise TypeError("WARINING: illegal Beamtime object argument!.\n"
+                        "input object {} has type = {}\n"
+                        "Please make sure you are hadning correct object\n"
```
Comment (sbillinge) [inline xpdacq/utils.py]: please correct typos.  Please don't submit things for review containing typos.

File: xpdacq/utils.py [hunk 4]
```diff
@@ -6,12 +6,41 @@
...
+
+def _check_obj(obj_name, error_msg=None):
+    """ function to check if an object exists in current namespace
+
+    Parameter
+    ---------
+    obj_name : str
+        object names in string format
+    error_msg : str
+        error msg when target object cant't be found in current
```
Comment (sbillinge) [inline xpdacq/utils.py]: typo

File: xpdacq/utils.py [hunk 5]
```diff
@@ -447,12 +447,12 @@ def _name_parser(self, name_str):
         return name_list  # [first, last]
 
     def _phase_parser(self, phase_str):
-        """ parser for filed with <chem formular>: <phase_amount>
+        """ parser for filed with <chem formula>: <phase_amount>
```
Comment (sbillinge) [inline xpdacq/utils.py]: field

**Comment by chiahaoliu** [comment]
close #310

**Pull request merged**